Fixes and improvements in mqtt samples

devbotas opened the pull request:
## Description
Added .editorconfig from dotnet/iot repo, so the style can now be automatically and effortlessly maintained. Then, cleaned and updated MQTT samples subfolder:

- Added very basic usage samples for Ethernet and WiFi boards.
- Added slightly more sophisticated sample which uses certificate.
- Removed Mosquitto.TemperatureGauge sample, because it was broken on both ends (bad certificates, impossible WiFi connections and also Gauge itself is broken on test.mosquitto.org).
- Updated readme accordingly. 
- Cleanup up samples folder, as there were some ghost files.

## Motivation and Context
There was no basic and easy samples, especially ones that actually work.

## How Has This Been Tested?
Tested on STM32F769 and ESP Pico boards.

## Screenshots<!-- (if appropriate): -->

## Types of changes
- [x] Improvement (non-breaking change that improves a sample)
- [x] Bug fix (fixes an issue with a current sample)
- [x] New Sample (adds a new sample)

## Checklist:
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

josesimoes commented on samples/MQTT/BasicExample.WiFi/app.config:
```diff
@@ -0,0 +1,15 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
```
app.config not supported please remove

josesimoes commented on README.md:
```diff
@@ -17,7 +17,7 @@ Feel free to browse, take what you like and contribute back if you want.
 | --- | --- | --- |
 | [SerialCommunication sample pack](samples/SerialCommunication) | [CAN sample](samples/CAN) | [AMQP sample pack](samples/AMQP) |
-| [MQTT sample pack](samples/MQTT/TestMqtt) | [MQTT sample with Azure IoT Hub](samples/MQTT/AzureMQTT) | [MQTT sample with Amazon Web Services (AWS) IoT](samples/MQTT/AwsMQTT) |
+| [MQTT sample pack](samples/MQTT) | | |
```
I would say that the explicit mention of both Azure IoT and AWS in the main readme increase the visibility of those samples and demonstrate that we support for both vendors.

josesimoes reviewed (commented):
Great that you've tidy up and update the MQTT samples. A few comments:

1. I would rather have the samples in separated solutions. Having them in a single one makes it hard for developers to grab the one that they are interested in and start making it's changes.

2. As this is being improved let's have more info to help new comers. There should be a readme with some pointers to Azure IoT, AWS and mosquito server on the respective folders.

josesimoes commented:
Requesting feedback from @networkfusion and @Ellerbach as mqtt sample owners 😉

networkfusion commented:
I would rather keep AWS and Azure solutions seperate, especially since I am still working on the AWS one in the background for device shadow support. Also, it is unlikely someone would be looking to support both cloud vendors at the same time (without a lot of work until my helper lib is working properly)...

Ellerbach commented:
> I would rather keep AWS and Azure solutions seperate, especially since I am still working on the AWS one in the background for device shadow support. Also, it is unlikely someone would be looking to support both cloud vendors at the same time (without a lot of work until my helper lib (https://github.com/networkfusion/OrgPal-Three-Demo/tree/develop-awsiotlib/OrgPalThreeDemo/AwsIoTtestLib) is working properly)...

+1 with separate. They are very specific. And they may change as well (they will with broder support of MQTT5), they are specific example and for sure some devs will prefer to use those than SDK as well.

devbotas commented:
Updated as per requests, mainly split the solution into multiple ones. Updated links in master readme.

AWS still uses preview libraries, and Azure does not build at all, but I'll leave those problems to the sample owner to fix.

Also, I did not quite get what "There should be a readme with some pointers to Azure IoT (...)" means, so I'll just leave PR as-is now and wait for it to be merged (or rejected).

Then maybe we'll be able to do another round.

josesimoes reviewed (approved):
LGTM!

josesimoes commented:
@Ellerbach @networkfusion when you have a few minutes, please review these changes.

Ellerbach reviewed (approved):
Looks good. thanks!

networkfusion reviewed (approved):
Definitely better than before, and a good step towards the future.
